.github/workflows/ci.yaml

```diff
@@ -0,0 +1,278 @@
+---
+name: ci
+
+"on":
+  push:
+    branches:
+      - 'main'
+  pull_request:
+
+env:
+  CARGO_INCREMENTAL: 0
+  # We are on an old nightly so we can't use sparse,
+  # remove this when we get past 2023-03-23 nightly
+  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: "git"
+
+jobs:
+  # TODO: Fix automatically
+  lint:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt
+      - uses: r7kamura/rust-problem-matchers@v1
+      # Hack: the rust-toolchain file confuses cargo fmt what version to use
+      - run: rm rust-toolchain && cargo fmt --all -- --check
```

Comment:
I just saw this, https://github.com/mobilecoinfoundation/cookiecutters/pull/52#discussion_r1147125714
I think the hiccup here is that in this repo the toolchain file is a symlink to the submodule so cargo fmt looks at it and fails until the submodule is visible

Reply:
i see thanks --

i still don't want to checkout the submodule if i can help it because it started formatting in mobilecoin.git

maybe there's a rustfmt.toml ignore setting i can use but i think that is nightly only